resources/js/components/fieldtypes/QueryBuilder.vue

```diff
@@ -243,6 +259,22 @@ export default {
         }
     },
 
+    computed: {
+        flattenedFields() {
+            if (this.fields.length === 0 || !this.fields[0].options) {
+                return this.fields;
+            }
+
+            const flattened = [];
+            this.fields.forEach(group => {
+                group.options.forEach(field => {
+                    flattened.push(field);
+                });
+            });
+            return flattened;
```

Comment:
Javascript has a function for this: `const flattened = this.fields.flatMap(group => group.options);`